embedded/servo_driver/servo_driver.cc

```diff
@@ -3,26 +3,50 @@
 #include <algorithm> // min
 #include <cmath> // floor
 #include <string.h> // memset
+#include <sstream>
+#include <iostream>
+#include <fstream>
+#include <yaml-cpp/yaml.h>
 
-ServoDriver::ServoDriver(char* dev) : pwm_driver_(dev) {}
 namespace {
-  constexpr int PWM_FREQUENCY = 333;
   // 850 to 2150 microseconds for the PWM width, specified by the servo
   constexpr int MIN_COUNTS = 1160;
   constexpr int MAX_COUNTS = 2935;
 }
 
+ServoDriver::ServoDriver(const int channel, PwmDriver &pwm_driver, const std::string &cfg_path) : channel_(channel), pwm_driver_(pwm_driver), config_path_(cfg_path) {}
+
 int ServoDriver::init() {
-  pwm_driver_.init();
-  pwm_driver_.set_pwm_freq(PWM_FREQUENCY);
-  pwm_driver_.enable_auto_increment(true);
+  try {
+    YAML::Node config = YAML::LoadFile(config_path_);
+    max_angle_ = config["max_angle"].as<int>();
+    calibrated_center_ = config["calibrated_center"].as<int>();
+    calibrated_max_ = config["calibrated_max"].as<int>();
+  } catch (YAML::BadFile e) {
+    max_angle_ = 20;
+    calibrated_center_ = 50;
+    calibrated_max_ = 100;
+  }
```

Comment:
Suggest just crashing if an exception is thrown, instead of some unplanned behavior